### ISSUE TITLE
Lists : "RemoveExistingContentTypes" does not work

### ISSUE THREAD

**Issue by Amysno** [SharePoint/PnP-Provisioning-Schema#144]
When provisioning a ListInstance, the "**RemoveExistingContentTypes**" attribute does not work.

Even if i've set the ListInstance's attribute "RemoveExistingContentTypes" to "TRUE" (even tried with "true" and "1"), I still have all the default content types on my list.

For exemple, if I create a new List coming from TaskList, bind this list instance with only one of my custom content type, and set the attribute "RemoveExistingContentTypes" to true, i still have the "Task" content type in my list instance.

Here is a screenshot  of the content types bound to my list. "Actions" is my custom one, the right one. "Element" is the default CT for task lists.
![parametres - liste - internet explorer_2](https://cloud.githubusercontent.com/assets/5112089/20840523/cf9c33d2-b8b0-11e6-995a-eaa35afd42b6.jpg)

Here is my XML Template : 
```
<?xml version="1.0" encoding="utf-8" ?>
<pnp:Provisioning xmlns:pnp="http://schemas.dev.office.com/PnP/2016/05/ProvisioningSchema"
                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                  xsi:schemaLocation="http://schemas.dev.office.com/PnP/2016/05/ProvisioningSchema https://raw.githubusercontent.com/OfficeDev/PnP-Provisioning-Schema/master/OfficeDevPnP.ProvisioningSchema/ProvisioningSchema-2016-05.xsd">
  <pnp:Preferences></pnp:Preferences>
  <pnp:Templates>
    <pnp:ProvisioningTemplate ID="Gal_SiteListsProvision">
      <pnp:Lists>
 <pnp:ListInstance Title="Actions" Description="actions" DocumentTemplate="" TemplateType="171" Url="Lists/Actions" MinorVersionLimit="0" MaxVersionLimit="0" DraftVersionVisibility="0" TemplateFeatureID="f9ce21f8-f437-4f7e-8bc6-946378c850f0" RemoveExistingContentTypes="true" ContentTypesEnabled="true" EnableFolderCreation="false">
          <pnp:ContentTypeBindings>
            <pnp:ContentTypeBinding ContentTypeID="0x010800FEE83FB7878D9D488935E729B520D25E" Default="true" />
          </pnp:ContentTypeBindings>
          <pnp:Views>
            <View Name="{5825FAC0-A568-4E13-9FAC-9E20C9BA2823}" MobileView="TRUE" Type="HTML" DisplayName="Toutes les actions" Url="{site}/Lists/Actions/AllItems.aspx" Level="1" BaseViewID="5" ContentTypeID="0x" ImageUrl="/_layouts/15/images/issues.png?rev=44">
              <Query>
                <Where>
                  <Eq>
                    <FieldRef Name="responsable" />
                    <Value Type="Integer">
                      <UserID Type="Integer" />
                    </Value>
                  </Eq>
                </Where>
              </Query>
              <ViewFields>
                <FieldRef Name="Checkmark" />
                <FieldRef Name="ID" />
                <FieldRef Name="reference" />
                <FieldRef Name="LinkTitle" />
                <FieldRef Name="responsable" />
                <FieldRef Name="datelancement" />
                <FieldRef Name="dateprevueinitiale" />
                <FieldRef Name="dateprevuerevue" />
                <FieldRef Name="daterealisee" />
                <FieldRef Name="etataction" />
                <FieldRef Name="commentaire" />
              </ViewFields>
              <RowLimit Paged="TRUE">100</RowLimit>
              <Aggregations Value="Off" />
              <JSLink>hierarchytaskslist.js</JSLink>
            </View>
          </pnp:Views>
          <pnp:FieldRefs>
            <pnp:FieldRef ID="c3a92d97-2b77-4a25-9698-3ab54874bc6f" Name="Predecessors" DisplayName="Prédécesseurs" />
            <pnp:FieldRef ID="cd21b4c2-6841-4f9e-a23a-738a65f99889" Name="DueDate" DisplayName="Échéance" />
            <pnp:FieldRef ID="d2a04afc-9a05-48c8-a7fa-fa98f9496141" Name="RelatedItems" DisplayName="Éléments connexes" />
            <pnp:FieldRef ID="f4468cec-423d-4066-b755-7ecd6117f004" Name="action" DisplayName="Action" />
            <pnp:FieldRef ID="21393510-62df-4db1-9dbf-802d4844d7e8" Name="datelancement" DisplayName="Date de lancement" />
            <pnp:FieldRef ID="2522794a-1330-463a-8868-b2afe6368bcf" Name="dateprevueinitiale" DisplayName="Date prévue initiale" />
            <pnp:FieldRef ID="ee745205-8677-4f06-af0b-5f572ba33c1f" Name="dateprevuerevue" DisplayName="Date prévue revue" />
            <pnp:FieldRef ID="e1d3b20a-9c9a-44a0-903f-b129378156e0" Name="daterealisee" DisplayName="Date réalisée" />
            <pnp:FieldRef ID="f44f5ef5-ccda-4063-a25a-c70cb156558c" Name="etataction" DisplayName="Etat de l'action" />
            <pnp:FieldRef ID="b98928d2-4225-4964-9bed-247af39008e0" Name="reference" DisplayName="Référence" />
            <pnp:FieldRef ID="b3009db6-74ad-4a81-9a68-79baf9bbc59f" Name="responsable" Required="true" DisplayName="Responsable" />
            <pnp:FieldRef ID="00abc833-d413-410c-b2a3-c7ab154264de" Name="aujpluscinq" DisplayName="Deadline" />
          </pnp:FieldRefs>
        </pnp:ListInstance>
      </pnp:Lists>
    </pnp:ProvisioningTemplate>
  </pnp:Templates>
</pnp:Provisioning>
```

**Comment by PaoloPia** [comment]
Hi,
Thanks for your feedback. So far, we've not yet implemented that capability, which is defined at the schema level only. I create an issue (https://github.com/SharePoint/PnP-Sites-Core/issues/875) in the PnP-Sites-Core repository and will follow up there. I will close this one.

**Comment by PaoloPia** [comment]
Hi,
Double-checking the source come out that I did a mistake and the implementation is already there. I will check the issue and fix it, if any. Thanks.

**Comment by erwinvanhunen** [comment]
Hi!

I looked into this specific issue and the situation is as follows:

When you create a -new- list in a site, and the template of which the list is based upon contains Content Type references, then the RemoveExistingContentTypes switch will actually remove the content types.

However, when applying a template to a site where there is already a list present with the same URL as specified in the template we will update the list. However, as there might be list items in the list that use those content types, we cannot remove the content types. E.g. when updating an existing list we only allow adding content types, not removing all of them. I added a warning to the provisioning engine now that will show the moment we encounter such a situation.

I will close this issue again. If you feel the issue still stands, do not hesitate to reopen it in it's new location  https://github.com/SharePoint/PnP-Sites-Core/issues/875.
